portal/src/graphql/portal/GeneralSettings.tsx

```diff
@@ -0,0 +1,69 @@
+import React, { useCallback, useContext, useMemo, useState } from "react";
+import { TextField } from "@fluentui/react";
+import { Context } from "@oursky/react-messageformat";
+import deepEqual from "deep-equal";
+
+import ButtonWithLoading from "../../ButtonWithLoading";
+import NavigationBlockerDialog from "../../NavigationBlockerDialog";
+
+import styles from "./GeneralSettings.module.scss";
+
+interface State {
+  appName: string;
+}
+
+const GeneralSettings: React.FC = function GeneralSettings() {
+  const { renderToString } = useContext(Context);
+
+  const initialState = useMemo<State>(() => {
+    return {
+      appName: "",
+    };
+  }, []);
+
+  const [state, setState] = useState<State>(initialState);
+
+  const isFormModified = useMemo(() => {
+    return !deepEqual(initialState, state, { strict: true });
+  }, [initialState, state]);
+
+  const onAppNameChange = useCallback((_event, value?: string) => {
+    if (value === undefined) {
+      return;
+    }
+    setState((prevState) => ({
+      ...prevState,
+      appName: value,
+    }));
+  }, []);
+
+  const onFormSubmit = useCallback((ev: React.SyntheticEvent<HTMLElement>) => {
+    ev.preventDefault();
+    ev.stopPropagation();
+
+    alert("App name cannot be saved currently");
+  }, []);
+
+  return (
+    <form className={styles.root} onSubmit={onFormSubmit}>
+      <TextField
+        className={styles.textField}
+        label={renderToString("GeneralSettings.app-name.label")}
+        value={state.appName}
+        onChange={onAppNameChange}
+      />
```

Comment:
For now, can we just show the app ID in a disabled text field? So the save button can also be always disabled.

Reply:
ok